### PR TITLE
api: introduce WiP file-level annotations.

### LINKED ISSUE #9769: Support vNalpha in protoxform and API boosting
Our current workflow only supports v3 major as a target version, but folks will be wanting to merge alpha APIs ASAP (both cache and Wasm). This issue tracks adding in the requisite support for this.

### PR DESCRIPTION
**Pull request by htuch**
This is the new style for indicating a file is WiP and subject to
breaking changes. Rather than rely on alpha major versions, which are
coarse grained and introduce migration difficulties for operators, we
use a file-level annotation.

Risk level: Low
Testing: API/docs build, manual inspection of docs.

Fixes #9769.

Signed-off-by: Harvey Tuch <htuch@google.com>

### CODE REVIEW

**Comment by htuch** [comment]
CC @markdroth @dfawley @ejona86 FYI, hopefully not too controversial given that we agreed that well telegraphed API instability is OK.

**Comment by dfawley** [comment]
> FYI, hopefully not too controversial given that we agreed that well telegraphed API instability is OK.

SGTM, thank you for the heads-up.

File: api/envoy/config/transport_socket/tap/v2alpha/tap.proto [hunk 1]
```diff
@@ -6,13 +6,15 @@ import "envoy/api/v2/core/base.proto";
...
 
 import "udpa/annotations/migrate.proto";
+import "udpa/annotations/status.proto";
 import "validate/validate.proto";
 
 option java_package = "io.envoyproxy.envoy.config.transport_socket.tap.v2alpha";
 option java_outer_classname = "TapProto";
 option java_multiple_files = true;
 option (udpa.annotations.file_migrate).move_to_package =
     "envoy.extensions.transport_sockets.tap.v3";
```
Comment (lizan) [inline api/envoy/config/transport_socket/tap/v2alpha/tap.proto]: We can use this annotation altogether with the status, we can easily achieve `v3alpha` package for files subject to breaking change.

**Comment by mattklein123** [comment]
@lizan is going to take over this change so I'm going to close this out and he can open a new PR. Thank you!

**Pull request closed without merging**